### PR TITLE
add search box to roll inspector dialog

### LINKED ISSUE #6670: Feature Request: Filter for the roll inspector
Minor thing, but it would help to find what you want to predicate on when looking through the dozens of roll options if there was a filter for that list. Not really an issue once you're familiar with the possible options, but for someone new to the system, finding their way around, it'd be a nice-to-have, I think.

### PR DESCRIPTION
**Pull request by asuffield**
fixes #6670

### CODE REVIEW

**Comment by CarlosFdez** [comment]
Fix the conflicts when you get a chance, but also show a video of this feature so we know what exactly we're looking at.

File: src/module/chat-message/chat-roll-details.ts [hunk 1]
```diff
@@ -16,17 +22,62 @@ class ChatRollDetails extends Application {
...
+    private sortRollOptions() {
+        this.rollOptions.sort((a, b) => {
+            // Slightly weird js syntax: convert booleans to integers to generate a sort order from them.
+            return +a.includes(":") - +b.includes(":") || a.localeCompare(b);
+        });
+    }
+
+    override activateListeners($html: JQuery): void {
+        const textFilter = $html.find<HTMLInputElement>(".roll-options input")[0];
+        const optionsList = $html.find<HTMLInputElement>(".roll-options ul")[0];
```
Comment (stwlam) [inline src/module/chat-message/chat-roll-details.ts:62]: no jquery

**Comment by CarlosFdez** [comment]
Sorry, didn't see the post. My bad.

The only thing is that I'd rather if the roll options list were rendered by a partial, and partial was used in both the initial rendering and to render the search results. Other than that and what shark said, looks cool.

**Comment by CarlosFdez** [comment]
If you can come back to this let us know and we'll reopen it. For now to clear some room I'm going to close it.

**Pull request closed without merging**